src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Aes.cs

```diff
@@ -33,6 +35,470 @@ protected Aes()
             return (Aes?)CryptoConfig.CreateFromName(algorithmName);
         }
 
+        /// <summary>
+        ///   Computes the output length of the IETF RFC 5649 AES Key Wrap with Padding
+        ///   Algorithm for the specified plaintext length.
+        /// </summary>
+        /// <param name="plaintextLengthInBytes">
+        ///   The length of the plaintext to be wrapped, in bytes.
+        /// </param>
+        /// <returns>
+        ///   The padded length of the key wrap for the specified plaintext.
+        /// </returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="plaintextLengthInBytes"/> is less than or equal to zero.
+        /// </exception>
+        public static int GetKeyWrapPaddedLength(int plaintextLengthInBytes)
+        {
+            if (plaintextLengthInBytes <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(plaintextLengthInBytes),
+                    "Plaintext length must be positive.");
+            }
+
+            checked
+            {
+                int blocks = (plaintextLengthInBytes + 7) / 8;
+                return (blocks + 1) * 8;
+            }
+        }
+
+        /// <summary>
+        ///   Wraps a key using the IETF RFC 5649 AES Key Wrap with Padding algorithm.
+        /// </summary>
+        /// <param name="plaintext">The data to wrap.</param>
+        /// <returns>The wrapped data.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="plaintext"/> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentException"><paramref name="plaintext"/> is empty.</exception>
+        /// <exception cref="CryptographicException">An error occurred during the cryptographic operation.</exception>
+        public byte[] EncryptKeyWrapPadded(byte[] plaintext)
+        {
+            ArgumentNullException.ThrowIfNull(plaintext);
+
+            return EncryptKeyWrapPadded(new ReadOnlySpan<byte>(plaintext));
+        }
+
+        /// <summary>
+        ///   Wraps a key using the IETF RFC 5649 AES Key Wrap with Padding algorithm.
+        /// </summary>
+        /// <param name="plaintext">The data to wrap.</param>
+        /// <returns>The wrapped data.</returns>
+        /// <exception cref="ArgumentException"><paramref name="plaintext"/> is empty.</exception>
+        /// <exception cref="CryptographicException">An error occurred during the cryptographic operation.</exception>
+        public byte[] EncryptKeyWrapPadded(ReadOnlySpan<byte> plaintext)
+        {
+            if (plaintext.IsEmpty)
+                throw new ArgumentException(SR.Arg_EmptySpan, nameof(plaintext));
```

Comment:
Since the array method defers here, `EncryptKeyWrapPadded(Array.Empty<byte>())` throws "Span may not be empty." which is a tad confusing.